### DISCUSSION TITLE
rand.test custom test-statisitc (slope)

### DISCUSSION THREAD

**Discussion by Paulinevg** [jazznbass/scan#148] [Q&A] [answered]
Hello, 

I am trying to do a randomisation test on a multiple baseline design with the rand.test function. 
The test statistic I would like to use is difference in slope between phase A and B. 
However, this is not a standard test-statisitc in the package. Is it possible somehow to make a custom test-statistic? 

Thank you in advance!

**Comment by jazznbass** [comment]
Hi there,
I am currently working on a scan update which includes various new test statistics (slope difference, t-Value, nap und w-value).
You can try this new features out in the development version of scan (0.61.0).

see here how to install this version:

https://jazznbass.github.io/scan-Book/ch_scan_package.html#development-version-of-scan

best, jürgen

**Reply by jazznbass** [reply, marked as answer]
Meanwhile scan version 0.61.0 is on CRAN

**Reply by Paulinevg** [reply]
Hi, 

That is great, thank you!
I have been working with it over the past week, and it is very helpful. However, I have encountered a few problems that I have not been able to solve. 

1. If I have missing values (NA) I cannot run the rand_test. it states "invalid number of 'breaks'. 
2. I tried to solve this by adjusting the "mt" in my excel file (so for example if value on timepoint 3 misses, the mt colum would be (1 2 4 5 etc).  However, I think the "startpoints" in the rand_test function are not based on mt, but on the acual number of datapoints. I have multiple participants in my mbd, and because of missing values the possible starting points (based on the actual number of datapoints) is now different for everyone because of missing values. 
3.  A question: I want to investigate delayed effects. Is that possible with this package? If not, is it somehow possible to get all the t statistics for every possible "split" of the dataset? That way I would be able to do this by hand. 

If you could possibly help out with these things, that would be great. 
Thank you very much.

**Reply by jazznbass** [reply]
This is a good idea. I added the starting points for each permutation to the return object of the rand_test (`distribution_startpoints`).
And I added a new type to the plot_rand function.

```r
exampleAB[2] |> 
  rand_test(complete = TRUE, limit = 1, statistic = "SMD glass") |> 
  plot_rand(type = "xy")
```
![grafik](https://github.com/user-attachments/assets/9b4c936f-dbbc-404b-b707-ea238a96dde4)

The blue dot depict the statistic for the actual start of phase B

I uploaded this to the dev version of scan. Probably, the syntax might change a bit.

Additionally, I extended the rand_test function to include user made statistics with the  argument `statistic_function`:

```r
userstat <- list(
  statistic = function(a, b) median(b) - median(a), 
  aggregate = function(x) median(x),
  name = "median B - A"
)

rand_test(exampleAB, statistic_function = userstat , complete = TRUE)

# which is identical to:
rand_test(exampleAB, statistic = "Median B-A" , complete = TRUE)
```

**Reply by jazznbass** [reply]
Could you give a data example for your problem with missing values? (use the `convert()` function for an easy way)
